src/controller/SetUpCodePairer.cpp

```diff
@@ -104,55 +88,50 @@ CHIP_ERROR SetUpCodePairer::PairDevice(NodeId remoteId, const char * setUpCode,
     return errorCode;
 }
 
-CHIP_ERROR SetUpCodePairer::Connect(SetupPayload & payload)
+CHIP_ERROR SetUpCodePairer::Connect()
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-    bool isRunning = false;
-
-    bool searchOverAll = !payload.rendezvousInformation.HasValue();
-
     if (mDiscoveryType == DiscoveryType::kAll)
     {
-        if (searchOverAll || payload.rendezvousInformation.Value().Has(RendezvousInformationFlag::kBLE))
+        if (ShouldDiscoverUsing(RendezvousInformationFlag::kBLE))
         {
-            if (CHIP_NO_ERROR == (err = StartDiscoveryOverBLE(payload)))
+            CHIP_ERROR err = StartDiscoveryOverBLE();
+            if (err != CHIP_NO_ERROR)
             {
-                isRunning = true;
+                ChipLogError(Controller, "Failed to start discovery over BLE: %" CHIP_ERROR_FORMAT, err.Format());
             }
-            VerifyOrReturnError(searchOverAll || CHIP_NO_ERROR == err || CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE == err, err);
         }
 
-        if (searchOverAll || payload.rendezvousInformation.Value().Has(RendezvousInformationFlag::kSoftAP))
+        if (ShouldDiscoverUsing(RendezvousInformationFlag::kSoftAP))
         {
-            if (CHIP_NO_ERROR == (err = StartDiscoveryOverSoftAP(payload)))
+            CHIP_ERROR err = StartDiscoveryOverSoftAP();
+            if (err != CHIP_NO_ERROR)
             {
-                isRunning = true;
+                ChipLogError(Controller, "Failed to start discovery over SoftAP: %" CHIP_ERROR_FORMAT, err.Format());
             }
-            VerifyOrReturnError(searchOverAll || CHIP_NO_ERROR == err || CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE == err, err);
         }
-        if (searchOverAll || payload.rendezvousInformation.Value().Has(RendezvousInformationFlag::kWiFiPAF))
+        if (ShouldDiscoverUsing(RendezvousInformationFlag::kWiFiPAF))
         {
-            ChipLogProgress(Controller, "WiFi-PAF: has RendezvousInformationFlag::kWiFiPAF");
-            if (CHIP_NO_ERROR == (err = StartDiscoveryOverWiFiPAF(payload)))
+            ChipLogProgress(Controller,
```

Comment:
I realize we preserve logging here (and correct it), however why is this one-off? why do we log wifi-paf but not others?

It feels to me like this was "the new feature" and a logging path was added for that, however that makes the code inconsistent. I would rather drop this log or log the bitmaps somewhere or log consistently in every code path. Making wifipaf special is odd.